File: Services/MiratDestinationStore.cs
using System.IO;
using System.Text.Json;
using iPhotoImporter.Models;

namespace iPhotoImporter.Services;

/// <summary>
/// Persistència a disc de la llista de destins Mirat configurats.
/// Fitxer: %LocalAppData%\iPhotoImporter\mirat-destinations.json
/// </summary>
public class MiratDestinationStore
{
    private readonly string _filePath;
    private static readonly JsonSerializerOptions _jsonOptions = new()
    {
        WriteIndented = true,
        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
    };

    public MiratDestinationStore()
    {
        var folder = Path.Combine(
            Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
            "iPhotoImporter");
        Directory.CreateDirectory(folder);
        _filePath = Path.Combine(folder, "mirat-destinations.json");
    }

    /// <summary>Carrega la llista (buida si el fitxer no existeix o està malmès).</summary>
    public List<MiratDestination> Load()
    {
        if (!File.Exists(_filePath)) return [];
        try
        {
            var json = File.ReadAllText(_filePath);
            var items = JsonSerializer.Deserialize<List<MiratDestination>>(json, _jsonOptions);
            return items ?? [];
        }
        catch
        {
            return [];
        }
    }

    /// <summary>Desa la llista sencera (reemplaça el fitxer).</summary>
    public void Save(IEnumerable<MiratDestination> destinations)
    {
        try
        {
            var json = JsonSerializer.Serialize(destinations, _jsonOptions);
            File.WriteAllText(_filePath, json);
        }
        catch (Exception ex)
        {
            // No és crític: es perdrà la configuració però l'app continuarà.
            System.Diagnostics.Debug.WriteLine($"[MiratDestinationStore] Save failed: {ex.Message}");
        }
    }
}

File: MacOS/iPhotoManager/Services/MiratDestinationStore.swift
import Foundation

/// Persistència a disc de la llista de destins Mirat configurats.
/// Fitxer: ~/Library/Application Support/iPhotoManager/mirat-destinations.json
final class MiratDestinationStore {

    private let fileURL: URL

    private let encoder: JSONEncoder = {
        let e = JSONEncoder()
        e.outputFormatting = [.prettyPrinted, .sortedKeys]
        return e
    }()

    private let decoder = JSONDecoder()

    init() {
        let fm = FileManager.default
        let base = (try? fm.url(for: .applicationSupportDirectory,
                                in: .userDomainMask,
                                appropriateFor: nil,
                                create: true))
            ?? URL(fileURLWithPath: NSHomeDirectory()).appendingPathComponent("Library/Application Support")
        let folder = base.appendingPathComponent("iPhotoManager", isDirectory: true)
        try? fm.createDirectory(at: folder, withIntermediateDirectories: true)
        self.fileURL = folder.appendingPathComponent("mirat-destinations.json")
    }

    /// Carrega la llista (buida si el fitxer no existeix o està malmès).
    func load() -> [MiratDestination] {
        guard FileManager.default.fileExists(atPath: fileURL.path) else { return [] }
        do {
            let data = try Data(contentsOf: fileURL)
            return try decoder.decode([MiratDestination].self, from: data)
        } catch {
            return []
        }
    }

    /// Desa la llista sencera (reemplaça el fitxer).
    func save(_ destinations: [MiratDestination]) {
        do {
            let data = try encoder.encode(destinations)
            try data.write(to: fileURL, options: .atomic)
        } catch {
            // No és crític: es perdrà la configuració però l'app continuarà.
        }
    }
}
